### PR TITLE
fix: Request intercept response with multiple headers

### PR DESCRIPTION
**Pull request by jake314159**
Fixes the issue: https://github.com/GoogleChrome/puppeteer/issues/4379

### CODE REVIEW

**Comment by jake314159** [comment]
I signed it

File: lib/NetworkManager.js [hunk 1]
```diff
@@ -722,8 +722,14 @@ class SecurityDetails {
  */
 function headersArray(headers) {
   const result = [];
-  for (const name in headers)
-    result.push({name, value: headers[name]});
+  for (const name in headers) {
+    if (Array.isArray(headers[name])) {
+      for (let i = 0; i < headers[name].length; i++)
```
Comment (JoelEinbinder) [inline lib/NetworkManager.js]: nit: `for (const value of headers[name]) {`

**Comment by JoelEinbinder** [comment]
Thanks for the PR! Can you add a test?

**Comment by jasonswearingen** [comment]
@JoelEinbinder the test contents can be super simple, like:

```javascript


	const browser = await puppeteer.launch();
	const page = await browser.newPage();
	await page.setRequestInterception( true );

	page.on( "request", ( request ) => {

		if ( request.isNavigationRequest() !== true || page.mainFrame() !== request.frame() ) {
			request.continue();
		}
		request.respond( {
			body: "<html><body><h1>Hello</h1>World</body></html>",
			headers: { "Set-Cookie": [ "testCookie1=first-of2testcookies", "testCookie2=secondoftwotestcookies" ] }			
		} );
	} );


	const response = await page.goto( "http://example.com"	, { timeout: 6000, waitUntil: [] } );
```

Hope that helps you guys get this merged.....


***EDIT***: oops, I see @jake314159 already added a test and committed.... nevermind....

**Comment by aslushnikov** [comment]
Closing this as per discussion in the related issue.

**Pull request closed without merging**